docs/_docs/contributing/testing.md

```diff
@@ -127,17 +127,6 @@ $ sbt
 > testCompilation --help
 ```
 
-### Joint and separate sources compilation
-
-When the sources of a test consist of multiple source files places in a single directory they are passed to the compiler in a single run and the compiler decides in which order to compile them. In some cases, however, to reproduce a specific test scenario it might be necessary to compile the source files in several steps in a specified order. To achieve that one can add a `_${step_index}` suffix to a file name (before the `.scala` or `.java` extension) indicating the order of compilation. E.g. if the test directory contains files named `Foo_1.scala`, `Bar_2.scala` and `Baz_2.scala` then `Foo_1.scala` will be compiled first and after that `Bar_2.scala` together with `Baz_2.scala`.
```

Comment:
This section should probably be preserved.

Reply:
I also noticed the `JAVA_ONLY` convention for `.java`; I added `SCALA_ONLY` in a PR. I was about to convert that convention to header directives, such as `// test: --javac-only --scalac-only`. If that happens, I'll try to remember to document it. I wonder if `// test: --group:2` or similar would be easier or harder. Renaming test files is low on my list of favorite things, I mean it is last not first.